metrics-scraper/monitoring.go

```diff
@@ -159,33 +226,48 @@ func nodeVersion(ctx context.Context, client *rpc.Client, moniker, networkID str
 	return nil
 }
 
-func sidecarsVersion(ctx context.Context, client *rpc.Client, moniker, networkID string) error {
+func sidecarsVersion(ctx context.Context, client *rpc.Client, moniker, networkID string) (err error) {
+	defer func() {
+		if err != nil {
+			ethereumSidecarGauge.WithLabelValues(moniker, networkID, "0.0.0-unknown").Set(0)
+			connectSidecarGauge.WithLabelValues(moniker, networkID, "0.0.0-unknown").Set(0)
+		}
+	}()
+
+	// always remove the previous metrics to ensure that there's no duplicates when
+	// the labels version changes
+	ethereumSidecarGauge.DeletePartialMatch(map[string]string{
+		"moniker":  moniker,
+		"chain_id": networkID,
+	})
+	connectSidecarGauge.DeletePartialMatch(map[string]string{
+		"moniker":  moniker,
+		"chain_id": networkID,
+	})
+
 	result := map[string]net.SidecarInfos{}
-	err := client.CallContext(ctx, &result, netSidecarsEndpoint)
+	err = client.CallContext(ctx, &result, netSidecarsEndpoint)
 	if err != nil {
+		// just format nicely the error, and continue execution
+		// so the prometheus variable are still set to defaults
 		return fmt.Errorf("couldn't call %v for %v: %v", netSidecarsEndpoint, moniker, err)
```

Comment:
We use named return so let's be consistent:
```suggestion
        err = fmt.Errorf("couldn't call %v for %v: %v", netSidecarsEndpoint, moniker, err)
		return 
```

Reply:
Sure. I don't think they can't be compatible though. Here in those functions I use a named return to show the intent in the defer that we are checking te returned value explicit. Now, actually returning explicitly or not in the function is really just a matter of style. Happy to update the code to your preferred style though.

Reply:
addressed in: 94e0597d570fbb1a694bc047ed7b89d8570a1289